### PR TITLE
[docs] Add a Review section with overview in Home

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Closes ENG-10811

# How

- Add a "Review" section in the Home tab of docs by updating `navigation.js`
- Add an "Overview" page for the "Review" section that covers Internal distribution and Using EAS Update with development builds and provide links further to the relevant guides for more information.
- Add "How to launch an update using Expo Orbit" that describes requirements and steps to use Expo Orbit to launch an update from Expo dashboard.
- Update the `Video` component to add a caption (similarly to how we do with `ImageSpotlight`)

# Test Plan

Run docs locally and visit: http://localhost:3002/review/overview/ or see /review/overview/

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/review/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+<BoxLink
+  title="How to set up an internal distribution build"
+  description="Learn how EAS Build provides shareable URLs for your builds with your team for internal distribution."
+  href="/build/internal-distribution/"
+/>
+
+## Using EAS Update with development builds
+
+Development builds are highly recommended for a robust and flexible solution, especially when planning to deploy your app to production. They provide a comprehensive development environment that can be further configured to generate previews for your app during the review stage.
```
Comment (brentvatne) [inline docs/pages/review/overview.mdx:31]:
```suggestion
You can use [development builds](/develop/development-builds/introduction/) to load previews of your app during the review stage by publishing the preview with EAS Update.
```

File: docs/pages/review/overview.mdx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Even though Expo Go is an open-source sandbox that can be good for previewing isolated prototypes on Android and iOS, it is not intended for production apps. It should be avoided during the preview process of your app.
+
+You can use a development build with EAS Update and see the getting started guide for further instructions to set up the `expo-updates` library and learn how to create a link between a build and an update.
+
```
Comment (brentvatne) [inline docs/pages/review/overview.mdx:36]:
```suggestion
Even though Expo Go is an open-source sandbox that can be good for previewing isolated prototypes on Android and iOS, it is not intended for production apps. It should be avoided during the preview process of your app.
```

Comment (brentvatne) [reply docs/pages/review/overview.mdx:36]: also, maybe move this Expo Go part inside of a collapsible? like "Can I use Expo Go for reviewing releases?"

File: docs/pages/review/overview.mdx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+EAS Update allows you to push updates to iterate faster with your team. You can publish a new update for your team to preview. This process allows any team member or reviewer with a registered device to review the latest version of your app by scanning a QR code.
+
+<BoxLink
+  title="Develop with EAS Update"
+  description="Learn how to iterate faster locally or with your team."
+  href="/eas-update/develop-faster/"
+/>
+
+If you have the `expo-dev-client` library installed in your project, your team can launch different app versions and preview a published update inside the development build. It supports loading published EAS updates through channels.
+
```
Comment (brentvatne) [inline docs/pages/review/overview.mdx:52]: maybe we should reword this slightly and move it to the "description" field for the box link below?

File: docs/pages/review/overview.mdx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+<BoxLink
+  title="Use expo-dev-client with EAS Update"
+  description="Learn how to use expo-dev-client library to preview a published EAS Update inside a development build."
+  href="/eas-update/expo-dev-client/"
+/>
+
+You can also use GitHub Actions to automate the process of publishing updates. It makes deploying updates consistent and fast, leaving you more time to develop your app.
+
```
Comment (brentvatne) [inline docs/pages/review/overview.mdx:60]: same re: moving to box link description

**Review by brentvatne — CHANGES_REQUESTED** [review]

**Comment by wschurman** [comment]
Curious why we went with "Review" as opposed to "Preview". When I think of "Review" in the context of mobile apps, I think of the app store review processes.

File: docs/pages/review/overview.mdx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  href="/eas-update/getting-started/"
+  Icon={LayersTwo02Icon}
+/>
+
+<BoxLink
+  title="Develop with EAS Update"
+  description="Learn how to use EAS Update to push changes to iterate faster with your team. This process allows any team member or reviewer with a registered device to review the latest version of your app by scanning the QR code."
+  href="/eas-update/develop-faster/"
+  Icon={LayersTwo02Icon}
+/>
```
Comment (brentvatne) [inline docs/pages/review/overview.mdx:58]: this link may not be particularly useful anymore after https://github.com/expo/expo/pull/27461 - wdyt @wschurman / @amandeepmittal?

Comment (amandeepmittal) [reply docs/pages/review/overview.mdx:58]: I just saw Will's PR and got the context. Will remove this link.

**Comment by amandeepmittal** [comment]
> Curious why we went with "Review" as opposed to "Preview". When I think of "Review" in the context of mobile apps, I think of the app store review processes.

Following the same pattern that we do for naming things as we do on expo.dev/home. Context I had was we avoid using "preview" as a term here. Recently, there was a discussion on Slack thread.

File: docs/pages/review/overview.mdx (hunk 5 above)
Comment (amandeepmittal) [reply docs/pages/review/overview.mdx]: Done ✅

File: docs/pages/review/with-orbit.mdx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+## Preview an update with Expo Orbit
+
+<Video
+  url="https://www.youtube.com/watch?v=l-Lx7BglNUE"
+  spaceAfter={50}
+  caption="Expo Orbit launching an update directly from Expo dashboard to an Android Emulator."
+/>
+
+Previewing with Expo Orbit requires you to have an update published. If you haven't published an update, see [Publish an update](/eas-update/publish-updates/) before following the steps in the next section.
```
Comment (amandeepmittal) [inline docs/pages/review/with-orbit.mdx:28]:
```suggestion
Previewing with Expo Orbit requires you to have an update published. If you haven't published an update, see [Publish an update](/eas-update/publish/) before following the steps in the next section.
```

File: docs/pages/review/overview.mdx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  title={
+    <>
+      Use <CODE>expo-dev-client</CODE> with EAS Update
+    </>
+  }
+  description={
+    <>
+      Learn how to use <CODE>expo-dev-client</CODE> in your project to launch different app versions
+      and preview a published update inside a development build. This library also supports loading
+      published EAS updates through channels.
```
Comment (amandeepmittal) [inline docs/pages/review/overview.mdx:80]:
```suggestion
      Learn how to use <CODE>expo-dev-client</CODE> in your project to launch different app versions
      and preview a published update inside a development build.
```

**Review by brentvatne — APPROVED** [review]

**Pull request merged**